docker/sagemaker/Dockerfile.sm

```diff
@@ -3,14 +3,15 @@
 
 ARG DEVICE=gpu
 ARG DGL_VERSION=2.3.0
+ARG SOURCE
 
-FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker as branch-gpu
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-gpu-py311-cu121-ubuntu20.04-sagemaker} as branch-gpu
 ENV dev_type=GPU
 ARG DGL_VERSION
 # Install DGL GPU version
 RUN pip3 install dgl==${DGL_VERSION}+cu121 -f https://data.dgl.ai/wheels/torch-2.3/cu121/repo.html && rm -rf /root/.cache
 
-FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-cpu-py311-ubuntu20.04-sagemaker as branch-cpu
+FROM ${SOURCE:-763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-cpu-py311-ubuntu20.04-sagemaker} as branch-cpu
```

Comment:
So now there will be a default value as `763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-training:2.3.0-cpu-py311-ubuntu20.04-sagemaker` here?

Reply:
Yes, if SOURCE is not provided the default value will be used which guarantees the old scripts work as well. For 0.5 we can remove this